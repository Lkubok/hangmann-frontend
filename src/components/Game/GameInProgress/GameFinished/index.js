import GameFinished from "./GameFinished";
export default GameFinished;
