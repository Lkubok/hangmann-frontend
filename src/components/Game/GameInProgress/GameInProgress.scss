@import "../../../scss/mixins.scss";
@import "../../../scss/variables.scss";

.game-container {
  position: relative;
}
