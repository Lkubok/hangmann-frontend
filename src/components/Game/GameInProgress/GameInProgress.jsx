import React, { Component } from "react";
import { connect } from "react-redux";
import {
  deleteGame,
  pressLetter,
  closeGame,
  clearGameParams
} from "../../../actions/gameActions";
import Tiles from "./Tiles";
import Keypad from "./Keypad/index";
import GameInfo from "./GameInfo";
import GameFinished from "./GameFinished/index";
import * as selectors from "../../../reducers/selectors";
import "./GameInProgress.scss";

export class GameInProgress extends Component {
  handleGameDelete = id => () => {
    const { userName, userEmail, userLevel, userLang } = this.props;
    this.props.deleteGame(id);
    this.props.clearGameParams(userName, userEmail, userLevel, userLang);
  };
  handleKeyPress = e => this.props.pressLetter(e.key, this.props.gameId);
  render() {
    const { gameId, quoteAuthor, isFinished } = this.props;
    return (
      <>
        <div
          className="game-container"
          onKeyPress={this.handleKeyPress}
          tabIndex="0"
        >
          {isFinished ? <GameFinished /> : null}
          <Tiles />
          <div className="quote-author">
            <div className="quote-item">Quote by:</div>
            <div className="quote-item">{quoteAuthor}</div>
          </div>
          <Keypad />
          <button onClick={this.handleGameDelete(gameId)}>Leave game</button>
        </div>
        <GameInfo />
      </>
    );
  }
}
const mapStateToProps = state => ({
  gameId: selectors.getGameId(state),
  quoteAuthor: selectors.getQuoteAuthor(state),
  isFinished: selectors.getIsFinished(state),
  userName: selectors.getGameUserName(state),
  userEmail: selectors.getUserEmail(state),
  userLevel: selectors.getGameLevel(state),
  userLang: selectors.getGameLang(state)
});

const mapDispatchToProps = {
  deleteGame,
  pressLetter,
  closeGame,
  clearGameParams
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(GameInProgress);
