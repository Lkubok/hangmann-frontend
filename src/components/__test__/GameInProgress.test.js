import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { GameInProgress } from "../Game/GameInProgress/GameInProgress";
Enzyme.configure({ adapter: new Adapter() });

describe("GameInProgress component", () => {
  beforeEach(() => {});
  it("Should match snapshot", () => {
    const wrapper = shallow(<GameInProgress />);
    expect(wrapper).toMatchSnapshot();
  });
});
