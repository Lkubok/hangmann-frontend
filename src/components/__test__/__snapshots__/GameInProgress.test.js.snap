// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameInProgress component Should match snapshot 1`] = `
<Fragment>
  <div
    className="game-container"
    tabIndex="0"
  >
    <Connect(Tiles) />
    <div
      className="quote-author"
    >
      <div
        className="quote-item"
      >
        Quote by:
      </div>
      <div
        className="quote-item"
      />
    </div>
    <Connect(Keypad) />
    <button
      onClick={[Function]}
    >
      Leave game
    </button>
  </div>
  <Connect(GameInfo) />
</Fragment>
`;
